superset-frontend/src/dashboard/components/IconButton.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { MouseEventHandler } from 'react';
+import Icons from 'src/components/Icons';
+import { SupersetTheme } from '@superset-ui/core';
+import * as AntdIcons from '@ant-design/icons/lib/icons';
+
+interface IconButtonProps {
+  icon: keyof typeof Icons | keyof typeof AntdIcons;
+  label?: string;
+  onClick: MouseEventHandler<HTMLDivElement>;
+}
+
+const IconButton = ({ icon, label, onClick }: IconButtonProps) => {
+  const Icon = Icons[icon];
+  return (
+    <div
+      tabIndex={0}
+      role="button"
+      css={(theme: SupersetTheme) => ({
```

Comment:
We still have that discussion to do about using the css approach. Would you like to organize that conversation? I am curious to here your opinions. Maybe we can open a thread with a wider audience in the chat

Reply:
@rusackas is currently working on the definition of the pattern. He will make a proposition for general discussion. I'll be happy to share my opinions then. I'll change this piece of code to use `styled` while this definition is being cooked 😉

Reply:
I really am... I promise. It has examples and everything... just need to finish it, and shop it around before I go publicizing it.